packages/mui-joy/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,203 @@
+  },
+})<{ ownerState: BreadcrumbsProps }>(({ theme, ownerState }) => {
+  return [
+    {
+      ...(ownerState.size === 'sm' && {
+        fontSize: theme.vars.fontSize.sm,
+        padding: '0.5rem',
+      }),
+      ...(ownerState.size === 'md' && {
+        fontSize: theme.vars.fontSize.md,
+        padding: '0.75rem',
+      }),
+      ...(ownerState.size === 'lg' && {
+        fontSize: theme.vars.fontSize.lg,
+        padding: '1rem',
+      }),
+      lineHeight: 1,
+    },
+    {
+      backgroundColor: theme.palette.background.body,
+    },
+  ];
+});
+
+const BreadcrumbsOl = styled('ol', {
+  name: 'MuiBreadcrumbs',
+  slot: 'Ol',
+  overridesResolver: (props, styles) => styles.ol,
+})<{ ownerState: BreadcrumbsProps }>({
+  display: 'flex',
+  flexWrap: 'wrap',
+  alignItems: 'center',
+  padding: 0,
+  margin: 0,
+  listStyle: 'none',
+});
+
+const BreadcrumbsSeparator = styled('li', {
+  name: 'MuiBreadcrumbs',
+  slot: 'Separator',
+  overridesResolver: (props, styles) => styles.separator,
+})<{ ownerState: BreadcrumbsProps }>({
+  display: 'flex',
+  userSelect: 'none',
+  marginLeft: 8,
+  marginRight: 8,
+});
+
+function insertSeparators(
+  items: React.ReactNode[],
+  className: string,
+  separator: React.ReactNode,
+  ownerState: BreadcrumbsProps,
+) {
+  return items.reduce((acc: React.ReactNode[], current: React.ReactNode, index: number) => {
+    if (index < items.length - 1) {
+      acc = acc.concat(
+        current,
+        <BreadcrumbsSeparator
+          aria-hidden
+          key={`separator-${index}`}
+          className={className}
+          ownerState={ownerState}
+        >
+          {separator}
+        </BreadcrumbsSeparator>,
+      );
+    } else {
+      acc.push(current);
+    }
+
+    return acc;
+  }, []);
+}
+
+const Breadcrumbs = React.forwardRef(function Breadcrumbs(inProps, ref) {
+  const props = useThemeProps<typeof inProps & BreadcrumbsProps>({
+    props: inProps,
+    name: 'MuiBreadcrumbs',
+  });
+
+  const {
+    children,
+    className,
+    component = 'nav',
+    size = 'md',
+    expandText = 'Show path',
+    separator = '/',
+    ...other
+  } = props;
+
+  const ownerState = {
+    ...props,
+    component,
+    expandText,
+    separator,
+    size,
+  };
+
+  const classes = useUtilityClasses(ownerState);
+
+  const listRef = React.useRef<HTMLOListElement>(null);
+
+  const allItems = React.Children.toArray(children)
+    .filter((child) => {
+      return React.isValidElement(child);
+    })
+    .map((child, index) => (
+      <li className={classes.li} key={`child-${index}`}>
+        {child}
+      </li>
+    ));
+
+  return (
+    <BreadcrumbsRoot
+      ref={ref}
+      className={clsx(classes.root, className)}
+      ownerState={ownerState}
+      {...other}
+    >
+      <BreadcrumbsOl className={classes.ol} ref={listRef} ownerState={ownerState}>
+        {insertSeparators(allItems, classes.separator, separator, ownerState)}
```

Comment:
The function is not that big and it is not complicated. I don't see a need to extract to the outer scope. It makes the code harder to read (jump back and forth).
```suggestion
        {items.reduce(...)}
```